### PR TITLE
AzureRT updates

### PR DESCRIPTION
**Pull request by hyonholee**
1. Change the warning message to a non-terminating error message for ResourceNotFound in VM cmdlets (ASM).
2. Add Id parameter for VM cmdlets (ARM) to support piping scenario without ResourceGroupName parameter.
3. Fix Set-AzureVMAccess extension 
4. Update Publish-AzurePlatformExtension and Set-AzurePlatformExtension cmdlets
5. Adding Add-AzureVhd cmdlet for ARM
#115
#369

### CODE REVIEW

**Comment by azurecla** [comment]
Hi __@hyonholee__, I'm your friendly neighborhood Azure Pull Request Bot (You can call me AZPRBOT). Thanks for your contribution!
    <p>
        It looks like you're working at Microsoft (hylee). If you're full-time, we DON'T require a contribution license agreement.
    </p>
    <p>
        If you are a vendor, or work for Microsoft Open Technologies, DO please sign the electronic contribution license agreement. It will take 2 minutes and there's no faxing! https://cla.azure.com.
    </p>

TTYL, AZPRBOT;

File: src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/VirtualMachineActionBaseCmdlet.cs [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+           ParameterSetName = ResourceGroupNameParameterSet,
+           ValueFromPipelineByPropertyName = true,
+           HelpMessage = "The resource group name.")]
+        [ValidateNotNullOrEmpty]
+        public string ResourceGroupName { get; set; }
+
+        [Parameter(
+           Mandatory = true,
+           Position = 0,
+           ParameterSetName = IdParameterSet,
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/VirtualMachineActionBaseCmdlet.cs]: If any child parameter derive from this base cmdlet class has another parameter set, will it have access to this parameter defined in ```IdParameterSet```?

**Comment by huangpf** [comment]
LGTM

**Comment by markcowl** [comment]
@hyonholee  Please pull the latest and resubmit

**Comment by markcowl** [comment]
@hyonholee Also, can you please submit changes to changelog.md for this and the previous PR?

File: src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/VirtualMachineActionBaseCmdlet.cs (hunk 1 above)
Comment (hyonholee) [inline src/ResourceManager/Compute/Commands.Compute/VirtualMachine/Action/VirtualMachineActionBaseCmdlet.cs]: That is an interesting question.  Currently all child cmdlets of this base method do not have an additional parameter set. I think a child cmdlet can have more parameter sets in addition to these parameter sets.

**Comment by markcowl** [comment]
@azuresdkci add to whitelist

**Comment by markcowl** [comment]
@azuresdkci retest this please

**Comment by markcowl** [comment]
@hyonholee Test failures

File: src/ResourceManager/Compute/Commands.Compute/StorageServices/AddAzureVhdCommand.cs [hunk 2]
```diff
@@ -0,0 +1,175 @@
...
+
+        [Parameter(
+            Position = 1,
+            Mandatory = true,
+            ValueFromPipelineByPropertyName = true,
+            HelpMessage = "Uri to blob")]
+        [ValidateNotNullOrEmpty]
+        [Alias("dst")]
+        public Uri Destination
+        {
```
Comment (huangpf) [inline src/ResourceManager/Compute/Commands.Compute/StorageServices/AddAzureVhdCommand.cs]: Just make it one line ```{ get; set; }```

**Comment by markcowl** [comment]
@azuresdkci retest this please

**Comment by markcowl** [comment]
on demand run is here: http://azuresdkci.cloudapp.net/view/1-AzurePowerShell/job/powershell-on-demand/159/

**Pull request merged**